### ISSUE TITLE
[desk-tool] Document title not resolving same values as list

### ISSUE THREAD

**Issue by rexxars** [sanity-io/sanity#418]
<img width="553" alt="screen shot 2017-12-09 at 19 37 05" src="https://user-images.githubusercontent.com/48200/33798464-d7ef8d44-dd18-11e7-96b6-7ecf1fec4f92.png">

The list titles resolve properly but document title does not seem to follow same logic.
Preview config:

```js
{
  preview: {
    select: {
      homeName: 'teams.home.name',
      awayName: 'teams.away.name',
      homeImage: 'teams.home.profileImage.asset.url',
      awayImage: 'teams.away.profileImage.asset.url', 
    },
    prepare: values => {
      return {title: `${values.homeName} vs ${values.awayName}`}
    }
  }
}
```

**Comment by bjoerge** [comment]
Still an issue?

**Comment by bjoerge** [comment]
I'm pretty sure this was fixed by #465. Reopen if still an issue.

**Comment by tyx** [comment]
Hi !

We have exactly this issue with an up to date sanity version:

![image](https://user-images.githubusercontent.com/245494/79019739-4b210000-7b77-11ea-9506-49e8c2462db2.png)

From what I see in console, nested object are not the same in document view.

```
// In list view console.log(debug): see code below
_id: "474e6090-4c82-4885-84d8-5746a95e56ca"
​_rev: "jJf2nTkqF10DuWZ1VvQBEa"
​_type: "unit"
​name: "201"
```


```
// In document view console.log(debug): see code below
_id: "474e6090-4c82-4885-84d8-5746a95e56ca"
​ref: "e3bd1133-80c8-49f3-95fc-9295ff398395"
_type: "reference"
```

Here is my schema
```js
import { FaNetworkWired } from "react-icons/fa";

const FilterRoom = ({ document }) => {
  if (!document.unitA && !document.unitB) {
    return {
      filter: "hotel._ref == $hotelId",
      params: {
        hotelId: document.hotel._ref,
      },
    };
  }

  return {
    filter: "hotel._ref == $hotelId && _id != $unitId",
    params: {
      hotelId: document.hotel._ref,
      unitId: document.unitA ? document.unitA._ref : document.unitB._ref,
    },
  };
};

export default {
  name: "unitConnection",
  title: "Unit Connection",
  icon: FaNetworkWired,
  type: "document",
  fields: [
    {
      name: "hotel",
      title: "Hotel",
      type: "reference",
      hidden: true,
      to: [{ type: "hotel" }],
      validation: Rule => Rule.required(),
    },
    {
      name: "unitA",
      title: "Unit A",
      type: "reference",
      to: [{ type: "unit" }],
      validation: Rule => Rule.required(),
      options: {
        filter: FilterRoom,
      },
    },
    {
      name: "unitB",
      title: "Unit B",
      type: "reference",
      to: [{ type: "unit" }],
      validation: Rule => Rule.required(),
      options: {
        filter: FilterRoom,
      },
    },
    {
      name: "pmsId",
      title: "PMS Identifier",
      type: "slug",
    },
  ],
  preview: {
    select: {
      a: "unitA.name",
      b: "unitB.name",
      debug: "unitA",
    },
    prepare(selection) {
      const { a, b, debug } = selection;
      console.log(debug);

      return {
        title: `${a} ⇆ ${b}`,
      };
    },
  },
};
```

What am I doing wrong ?
